apps/test/unit/componentLibrary/ToggleTest.jsx

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import sinon from 'sinon';
+
+import {expect} from '../../util/reconfiguredChai';
+
+import Toggle from '@cdo/apps/componentLibrary/toggle';
+describe('Design System - Toggle', () => {
+  it('Toggle - renders with correct label', () => {
+    render(
+      <Toggle name="test-toggle" value="test-toggle" label="Toggle label" />
+    );
+
+    const toggle = screen.getByDisplayValue('test-toggle');
+    expect(toggle).to.exist;
+    expect(screen.getByText('Toggle label')).to.exist;
+  });
+
+  it('Toggle - changes checked state on click', async () => {
+    const user = userEvent.setup();
+    const spyOnChange = sinon.spy();
+
+    let checked = false;
+    const onChange = () => {
+      checked = !checked;
+      spyOnChange(checked);
+    };
+
+    // Initial render
+    const {rerender} = render(
+      <Toggle
+        name="test-toggle"
+        value="test-toggle"
+        label="Toggle label"
+        checked={checked}
+        onChange={onChange}
+      />
+    );
+
+    let toggle = screen.getByDisplayValue('test-toggle');
+
+    expect(toggle).to.exist;
+    expect(toggle.checked).to.be.false;
+    expect(toggle.disabled).to.be.false;
+
+    await user.click(toggle);
+
+    // Re-render after user's first click
+    rerender(
+      <Toggle
+        name="test-toggle"
+        value="test-toggle"
+        label="Toggle label"
+        checked={checked}
+        onChange={onChange}
+      />
+    );
+
+    toggle = screen.getByDisplayValue('test-toggle');
+
+    expect(spyOnChange).to.have.been.calledOnce;
+    expect(spyOnChange).to.have.been.calledWith(true);
+    expect(toggle.checked).to.be.true;
+    expect(toggle.disabled).to.be.false;
+
+    await user.click(toggle);
+
+    // Re-render after user's second click
+    rerender(
```

Comment:
Docs [here say](https://testing-library.com/docs/react-testing-library/api/#rerender):

> It'd probably be better if you test the component that's doing the prop updating to ensure that the props are being updated correctly (see [the Guiding Principles section](https://testing-library.com/docs/guiding-principles)). That said, if you'd prefer to update the props of a rendered component in your test, this function can be used to update props of the rendered component.

What does the rerender do here?

Reply:
We're passing `checked` prop here which is changing when you click on the component. `rerender` needed here to rerender component and check if changes really occured.

Reply:
Something I'm not quite wrapping my head around, let me know if you can help me understand. I would have thought that React is responsible for re-rendering when props change. Why do we have to do it explicitly?

Reply:
This is a `@testing-library/react` `render` function specific syntax.
When using it - if you need to render component with updated props - you need to make a `rerender` manually (`rerender` is a function returned by `render` call). [example from docs](https://testing-library.com/docs/example-update-props) , [StackOverflow question](https://stackoverflow.com/questions/76471293/how-to-update-props-using-testing-library-react)